### PR TITLE
♻️Story ads: move config

### PR DESCRIPTION
**Pull request by calebcordry**
Part 1 of larger effort to move to  smaller classes and  have better testing.

Moves config related logic to its own class.

Part of #23710

### CODE REVIEW

**Comment by calebcordry** [comment]
closes #22000

File: extensions/amp-story-auto-ads/0.1/test/story-mock.js [hunk 1]
```diff
@@ -57,22 +57,34 @@ export class MockStoryImpl extends AMP.BaseElement {
...
+export function addStoryAutoAdsConfig(doc, parent, customConfig) {
   const config = customConfig || {
-    'ad-attributes': {
-      type: 'doubleclick',
-      'data-slot': '/30497360/a4a/fake_ad_unit',
-    },
+    type: 'doubleclick',
+    'data-slot': '/30497360/a4a/fake_ad_unit',
   };
   const child = doc.createElement('script');
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/test/story-mock.js]: probably more readable to turn the whole thing into an HTML blob

File: extensions/amp-story-auto-ads/0.1/story-ad-config.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+const AllowedAdTypes = {
+  'custom': true,
+  'doubleclick': true,
+  'fake': true,
+};
+
+export class StoryAdConfig {
+  /**
+   * @param {!Window} win
+   * @param {!Element} element
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/story-ad-config.js]: JS doc: amp-story element

File: extensions/amp-story-auto-ads/0.1/story-ad-config.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+   */
+  constructor(win, element) {
+    this.win_ = win;
+    this.element_ = element;
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/story-ad-config.js]: `@type`

File: extensions/amp-story-auto-ads/0.1/story-ad-config.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+    );
+
+    if (type === 'fake') {
+      const {id} = this.element_;
+      userAssert(
+        id && startsWith(id, 'i-amphtml-demo-'),
+        `${TAG} id must start with i-amphtml-demo- to use fake ads`
+      );
+    } else if (type === 'custom') {
+      Services.extensionsFor(this.win_)./*OK*/ installExtensionForDoc(
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/story-ad-config.js]: this logic does not naturally fall into a config parsing class, especially it causes a side-effect of DOM mutation. might belong to elsewhere.

File: extensions/amp-story-auto-ads/0.1/story-ad-config.js [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+      adAttributes,
+      requiredAttrs
+    ));
+  }
+
+  /**
+   * Logic specific to each ad type.
+   * @param {string} type
+   */
+  handleType_(type) {
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/story-ad-config.js]: `validateType_`

File: extensions/amp-story-auto-ads/0.1/story-ad-config.js [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+   */
+  getConfig() {
+    const child = this.element_.firstElementChild;
+    userAssert(
+      child && isJsonScriptTag(child),
+      `The ${TAG} should ` +
+        'be inside a <script> tag with type="application/json"'
+    );
+
+    const dirtyConfig = parseJson(child.textContent);
```
Comment (lannka) [inline extensions/amp-story-auto-ads/0.1/story-ad-config.js]: `dirtyConfig` => `jsonConfig`

**Pull request merged**